.changeset/many-worms-bow.md

```diff
@@ -0,0 +1,5 @@
+---
+'@urql/core': minor
+---
+
+fix maskTypename
```

Comment:
```suggestion
Fix case where `maskTypename` would not traverse down when the root query-field does not contain a `__typename`
```

Reply:
Thanks for your feedback ! 

I have added a unit test and fixed the case with Date objects